contracts/registries/IPAssetRegistry.sol

```diff
@@ -21,16 +24,32 @@ import { IPAccountStorageOps } from "../lib/IPAccountStorageOps.sol";
 ///         attribution and an IP account for protocol authorization.
 ///         IMPORTANT: The IP account address, besides being used for protocol
 ///                    auth, is also the canonical IP identifier for the IP NFT.
-contract IPAssetRegistry is IIPAssetRegistry, IPAccountRegistry {
+contract IPAssetRegistry is IIPAssetRegistry, IPAccountRegistry, AccessManagedUpgradeable, UUPSUpgradeable {
     using ERC165Checker for address;
     using Strings for *;
     using IPAccountStorageOps for IIPAccount;
 
+    /// @dev Storage structure for the IPAssetRegistry
     /// @notice Tracks the total number of IP assets in existence.
-    uint256 public totalSupply = 0;
+    /// @custom:storage-location erc7201:story-protocol.IPAssetRegistry
+    struct IPAssetRegistryStorage {
+        uint256 totalSupply;
+    }
+
+    // keccak256(abi.encode(uint256(keccak256("story-protocol.IPAssetRegistry")) - 1)) & ~bytes32(uint256(0xff));
+    bytes32 private constant IPAssetRegistryStorageLocation =
+        0x987c61809af5a42943abd137c7acff8426aab6f7a1f5c967a03d1d718ba5cf00;
 
+    /// @custom:oz-upgrades-unsafe-allow constructor
     constructor(address erc6551Registry, address ipAccountImpl) IPAccountRegistry(erc6551Registry, ipAccountImpl) {}
 
+    /// @notice Initializes the IPAssetRegistry contract.
+    /// @param accessManager The address of the access manager.
+    function initialize(address accessManager) public initializer {
```

Comment:
I saw there was a check if accessManager = address(0) in other PR.  Do you need to add this check here?
https://github.com/storyprotocol/protocol-core-v1/pull/60/files#diff-cd57b00a1247e3ce32fcf5e25ba6a371cecbee670026706be6563edca4bbec4aR96

Reply:
Fixed b020807d33f4dfe9c1fa5f14a0567f07624a2d9e